### PR TITLE
Experimental jbuilder-x

### PR DESCRIPTION
**Pull request by toots**
Well, that's quite exciting!

### CODE REVIEW

**Comment by whitequark** [comment]
The Travis build failed.

**Comment by toots** [comment]
Yeah, there's something odd going on with the script. I'll make sure to fix it when I'm done with the main changes.

**Comment by toots** [comment]
Ok, this is ready to roll! CI timed out so I ran the tests locally:
```
% ./tests/run_tests.sh | tee /tmp/log
Building pretest image..[ok]✅

From github.com:whitequark/opam-cross-windows
 * branch            master     -> FETCH_HEAD

Building base-windows.v0.9.3.. [ok]✅
Building camomile-windows.0.8.7.. [ok]✅
Building configurator-windows.v0.9.1.. [ok]✅
Building jbuilder-x.1.0+beta16.. [ok]✅
Building lwt-windows.2.6.0.. [ok]✅
Building lwt-windows.3.0.0.. [ok]✅
Building lwt_react-windows.1.0.1.. [ok]✅
Building ocaml-compiler-libs-windows.v0.9.0.. [ok]✅
Building ocaml-migrate-parsetree-windows.1.0.7.. [ok]✅
Building ocaml-windows32.4.04.0.. [ok]✅
[skipped]⚠️
Building ocamlbuild-windows.0.12.0.. [ok]✅
Building octavius-windows.1.2.0.. [ok]✅
Building pcre-windows.7.3.4.. [ok]✅
Building ppx_ast-windows.v0.9.1.. [ok]✅
Building ppx_base-windows.v0.9.0.. [ok]✅
Building ppx_compare-windows.v0.9.0.. [ok]✅
Building ppx_core-windows.v0.9.0.. [ok]✅
Building ppx_derivers-windows.1.0.. [ok]✅
Building ppx_driver-windows.v0.9.1.. [ok]✅
Building ppx_enumerate-windows.v0.9.0.. [ok]✅
Building ppx_hash-windows.v0.9.0.. [ok]✅
Building ppx_js_style-windows.v0.9.0.. [ok]✅
Building ppx_metaquot-windows.v0.9.0.. [ok]✅
Building ppx_optcomp-windows.v0.9.0.. [ok]✅
Building ppx_sexp_conv-windows.v0.9.0.. [ok]✅
Building ppx_traverse_builtins-windows.v0.9.0.. [ok]✅
Building ppx_type_conv-windows.v0.9.1.. [ok]✅
Building sexplib-windows.v0.9.2.. [ok]✅
Building stdio-windows.v0.9.1.. [ok]✅
```

**Comment by whitequark** [comment]
Can you rebase?

**Comment by whitequark** [comment]
Also, the commit message format is `+foo.1.2.3`, not `+foo.v1.2.3`, and for changes that touch only one package it's `foo.1.2.3: do something`. Thanks!

**Comment by toots** [comment]
Done. The `foo.vx.y.z` comes from the packages actual version. Cheers!

File: packages/jbuilder-x.1.0+beta16/files/jbuilder-x.install [hunk 1]
```diff
@@ -0,0 +1,4 @@
+bin: [
+  "jbuilder-x" {"jbuilder-x"}
```
Comment (whitequark) [inline packages/jbuilder-x.1.0+beta16/files/jbuilder-x.install]: This would conflict with a `jbuilder-x` from e.g. `opam-cross-android`.

Comment (toots) [reply packages/jbuilder-x.1.0+beta16/files/jbuilder-x.install]: `jbuilder-windows`?

Comment (whitequark) [reply packages/jbuilder-x.1.0+beta16/files/jbuilder-x.install]: Sounds good!

**Comment by toots** [comment]
`jbuilder-windows` is the best that I can find. Not totally a fan but I think that would be just fine.

**Comment by toots** [comment]
Hey @whitequark. I can't merge this because the CI times out. Not sure I want to break it into smaller PRs, the reason I pulled so many packages is that I wanted to get to compile `pcre-windows` as an example of a cross-build with `jbuilder` and an external library but turns out `pcre`'s dependencies are hairy.. And splitting into smaller PRs would bring the same headache..

**Pull request merged**